src/deadbot.py

```diff
@@ -37,6 +40,71 @@ def _update_page(self, page):
             print(f"No changes made to '{page.name}'")
 
 
-if __name__ == '__main__':
-    bot = DeadBot()
-    bot.run()
+def parse_arguments():
+    parser = argparse.ArgumentParser(
+        prog="DeadBot",
+        description="Bot that lives to serve deadlocked.wiki",
+        epilog="Process Deadlock game files and extract data and stats",
+    )
+
+    # Setup / Base config Flags
+    parser.add_argument(
+        "-i",
+        "--dl_path",
+        help="Path to Deadlock game files (also set with DEADLOCK_PATH environment variable)",
+        default=os.getenv("DEADLOCK_PATH"),
+    )
+    parser.add_argument(
+        "-w",
+        "--workdir",
+        help="Directory for temp working files (also set with WORK_DIR environment variable)",
+        default=os.getenv(
+            "WORK_DIR", os.path.abspath(os.getcwd()) + "/decompiled-data"
+        ),
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        help="Output directory (also set with OUTPUT_DIR environment variable)",
+        default=os.getenv("OUTPUT_DIR", os.path.abspath(os.getcwd()) + "/output-data"),
+    )
+    parser.add_argument(
+        "--decompiler_cmd",
+        help="Command for Valve Resource Format tool (also set with DECOMPILER_CMD env variable)",
+        default=os.getenv("DECOMPILER_CMD", "tools/Decompiler"),
+    )
+
+    # Operational Flags
+    parser.add_argument(
+        "-d",
+        "--decompile",
+        action="store_true",
+        help="Decompiles Deadlock game files. (also set with DECOMPILE environment variable)",
```

Comment:
Shall we default to use `DECOMPILE` here, similar to the other flags. As I noted in another comment, make sure you do checks for string bools

Reply:
Woops yep!  It just checked if it existed, not what it was.
Added a match against `[True, 'true', 'True', 'TRUE', 't', 'T', 1]`

Reply:
For the "action" parameters like decompile, I have them setup as flags so you can do `-d` instead of `-d=1`.
Because of that, the environment variables for those params are checked at runtime, which you found the bug in below!

It's setup so the user can use the parameter or env var, or a default value if one is sensible.
A truth table looks like this:
||env t|env f|
|param t| t| t|
|param f|t | default |

For all the "action" type flag params, they default to false or none.
For set parameters that have values, they default to hopefully good defaults